networking/src/NTP/Client.hs

```diff
@@ -146,14 +161,22 @@ startSend :: NtpMonad m => [SockAddr] -> NtpClient m -> m ()
 startSend addrs cli = do
     let timeout = ntpResponseTimeout (ncSettings cli)
     let poll    = ntpPollDelay (ncSettings cli)
-    logDebug "Sending requests"
-    liftIO . atomically . modifyTVarS (ncState cli) $ identity .= Just []
-    () <$ threadDelay timeout `race` forConcurrently addrs (flip doSend cli)
 
-    logDebug "Collecting responses"
-    handleCollectedResponses cli
-    liftIO . atomically . modifyTVarS (ncState cli) $ identity .= Nothing
-    liftIO $ threadDelay (poll - timeout)
+    _ <- concurrently (threadDelay poll) $ do
+        logDebug "Sending requests"
+        atomically . modifyTVarS (ncState cli) $ identity .= Just []
+        let sendRequests = forConcurrently addrs (flip doSend cli)
+        -- here we do only "send" part, so need to wait for some time
+        -- or till receiving all responses
+        let waitTimeout =
+                void $ race
+                    (threadDelay timeout)
+                    (atomically $ check =<< allResponsesGathered cli)
```

Comment:
Ok, so here we have a first-to-finish sync between a timeout and a (last-to-finish sync between all the responses). So this says our policy is to wait for all responses, but only up to the timeout. So in practise this means we must be prepared to wait for the timeout time in the typical case.

Reply:
It's assumed to communicate with small amount of reliable servers which should respond in reasonable time.
On the other hand, as it was mentioned, it may be better to poll many servers and wait for response from few ones to decrease probability of lagging on bad server.

Reply:
Can use `race_`

Reply:
`Mockable` module currently lacks `race_`, didn't want to touch extra modules in a hotfix